src/Symfony/Component/HttpKernel/Bundle/Bundle.php

```diff
@@ -191,6 +191,8 @@ public function registerCommands(Application $application)
             }
             $r = new \ReflectionClass($class);
             if ($r->isSubclassOf('Symfony\\Component\\Console\\Command\\Command') && !$r->isAbstract() && !$r->getConstructor()->getNumberOfRequiredParameters()) {
+                @trigger_error(sprintf('Auto-registration of the command "%s" is deprecated since Symfony 3.4 and won\'t be supported in 4.0. Use PSR-4 based Service Discovery instead.', $class), E_USER_DEPRECATED);
```

Comment:
enough indeed, registering the command as a service make convention based skipped thanks to the `console.command.ids` parameter